src/view/RestaurantView.java

```diff
@@ -105,6 +104,12 @@ public void propertyChange(PropertyChangeEvent evt) {
         info.add(restaurantAddress);
         info.add(restaurantPhoneNumber);
         info.add(restaurantCategories);
+
```

Comment:
If your action here involves toggling the visibility of the `addToFavorites` button, you can utilize `addToFavorites.setVisible(false)` inside `actionPerformed` to hide it, and similarly to show it again.